### PR TITLE
refactor: added get_config_dir()

### PR DESCRIPTION
**Pull request by CD3**
This commit consolidates config directory detection. There were a few places in the code that did the same thing, check if the user has XDG_CONFIG_HOME defined, and default so ~/.config if not, and I found that one of them had an error.

My original motivation was to add support for local configuration files so I can have groups of repos in separate directories and use gita on each group.

Added a function to return the application configuration directory and
replaced all manual calls to os.environ.get('XDG_CONFIG_HOME') with it.

Found three places that needed to load a configuration files from the
applications config directory (i.e. ~/.config/gita) two in utils.py
and one in info.py. The call two call in utils.py were correct, but
info.py had a bug.

The following code
root = os.environ.get('XDG_CONFIG_HOME') or os.path.join(
       os.path.expanduser('~'), '.config', 'gita')

src_fname = os.path.join(root, 'extra_repo_info.py')
yml_fname = os.path.join(root, 'info.yml')

Will result in src_fname = "~/.config/extra_repo_info.py" and yml_fname
= "~/.config/info.yml" if XDG_CONFIG_HOME is defined in the user
environment.

### CODE REVIEW

**Comment by nosarthur** [comment]
Thanks @CD3 for the PR. I will try to take a look over the weekend.

Also, please remove the swap files in the commit, thanks

**Comment by CD3** [comment]
Oops. Sorry, I always have them in my `.gitignore`. I removed them and committed, but would you prefer I squash into a single commit?

File: gita/common.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import os
+
+def get_config_dir() -> str:
```
Comment (nosarthur) [inline gita/common.py]: is it possible to have this funciton in `utils.py`?

File: gita/info.py [hunk 2]
```diff
@@ -42,8 +43,7 @@ def get_info_items() -> Tuple[Dict[str, Callable[[str], str]], List[str]]:
     display_items = ['branch', 'commit_msg']
 
     # custom settings
-    root = os.environ.get('XDG_CONFIG_HOME') or os.path.join(
-        os.path.expanduser('~'), '.config', 'gita')
```
Comment (nosarthur) [inline gita/info.py]: good catch, thanks!

File: gita/common.py (hunk 1 above)
Comment (nosarthur) [reply gita/common.py]: I just realized that `utils.py` imports `info.py` ...

Comment (CD3) [reply gita/common.py]: Yes. I originally was going to put it in `utils.py`

**Pull request merged**